TunnelKit/Sources/Core/RoutingTableEntry.m

```diff
@@ -243,15 +248,20 @@ - (BOOL)matchesDestination:(NSString *)destination
     NSMutableArray<RoutingTableEntry *> *segments = [[NSMutableArray alloc] init];
     const int halfPrefix = (int)(self.prefix + 1);
     if (self.isIPv6) {
+        if (self.prefix == 128) {
+            NSLog(@"Can't partition single IPv6");
+            return @[self, self];
+        }
+        
         struct in6_addr saddr1, saddr2;
         char addr[INET6_ADDRSTRLEN];
         NSData *addressData = RoutingTableEntryAddress6(self.network);
```

Comment:
NULL check of addressData?

```
if ( addressData == NULL ) return NULL;
```

Reply:
Also NULL checks after calls to ```RoutingTableEntryAddress6``` in ```matchesDestination```. At least in my fuzzers I observe crashes if those checks are not there.

```RoutingTableEntryAddress4``` calls might need similar checks.